src/code/z_camera.c

```diff
@@ -574,71 +574,86 @@ s16 Camera_XZAngle(Vec3f* to, Vec3f* from) {
     return CAM_DEG_TO_BINANG(RAD_TO_DEG(Math_FAtan2F(from->x - to->x, from->z - to->z)));
 }
 
-s16 func_80044ADC(Camera* camera, s16 yaw, s16 arg2) {
-    static f32 D_8015CE50;
-    static f32 D_8015CE54;
-    static CamColChk D_8015CE58;
+s16 Camera_GetPitchAdjFromFloorHeightDiffs_(Camera* camera, s16 viewYaw, s16 initAndReturnZero) {
```

Comment:
I'm trying to better follow the `initAndReturnZero` arg. Having an "init" arg makes me think you need to call it at least twice, once to init and then again for the main calcs.

`Camera_GetPitchAdjFromFloorHeightDiffs` is only called once in `Camera_Normal1` (default cam, false), once in `Camera_Normal3` (epona, true), once in `Camera_Parallel1` (z-parallel, true).

Is this behavior for when your switching between Normal

So what's the purpose of `initAndReturnZero = true`?
`sFloorYSmallOffsetForwards = sFloorYBigOffsetForwards = camera->playerGroundY;`
means
`floorYDiffSmallOffsetForwards` and `floorYDiffBigOffsetForwards` are both 0.

following the trig functions, yeah both `pitchSmallOffset` and `pitchBigOffset` end up 0. So the return being 0 checks out.

Is this `initAndReturnZero = true` behavior for when you're switching from `Normal3` -> `Normal1` or `Parallel1` -> `Normal1`?

Reply:
yeah I'm not sure why Normal3 and Parallel1 always pass `true`, probably for transition with Normal1 yeah but there were way better ways to implement that, the way this is done wastes a costly `Camera_BGCheckInfo` call when `initAndReturnZero = true` is that every frame
Also if it's just for initialization why do Normal3 and Parallel1 seemingly properly use the pitch result

So yeah idk, probably how the code got left on friday night and forgotten about when picked back up monday morning